### PR TITLE
EPMDJ-4918: Implement the ability to configure the session start as global and real-time

### PR DESCRIPTION
**Pull request by SMPlay**
Issue: EPMDJ-4918

### CODE REVIEW

File: src/modules/manage-sessions-pane/manage-new-session/manage-new-session.module.scss [hunk 1]
```diff
@@ -16,4 +16,28 @@
   &__new-session-form {
     padding: 16px 24px;
   }
+
+  &__custom-tooltip {
```
Comment (pavyarov) [inline src/modules/manage-sessions-pane/manage-new-session/manage-new-session.module.scss:20]: This is not ok. Plz contact design team. We have design system with tooltip. @RomanAchilov

File: src/types/start-session.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export interface StartSession {
```
Comment (pavyarov) [inline src/types/start-session.ts:1]: What is this?

**Review by pavyarov — CHANGES_REQUESTED** [review]

File: src/types/start-session.ts (hunk 2 above)
Comment (pavyarov) [reply src/types/start-session.ts:1]: StartSessionPayload at least

File: src/modules/manage-sessions-pane/manage-new-session/manage-new-session.tsx [hunk 3]
```diff
@@ -42,9 +48,51 @@ export const ManageNewSession = manageNewSession(({ className, agentId, serviceG
     </GeneralAlerts>
     <NewSessionForm label="Session ID">
       <Field name="sessionId" component={Fields.Input} placeholder="Enter session ID" />
+      <Checkboxes>
+        <Field
+          name="isGlobal"
+          type="checkbox"
+          render={({ input, meta }) => (
+            <Panel>
+              <Checkbox
```
Comment (pavyarov) [inline src/modules/manage-sessions-pane/manage-new-session/manage-new-session.tsx:57]: `Fields.Checkbox`

File: src/modules/manage-sessions-pane/manage-new-session/manage-new-session.tsx [hunk 4]
```diff
@@ -42,9 +48,51 @@ export const ManageNewSession = manageNewSession(({ className, agentId, serviceG
...
+                <IconInfo />
+              </Tooltip>
+            </Panel>
+          )}
+        />
+        <Field
+          name="isRealtime"
+          type="checkbox"
+          render={({ input, meta }) => (
+            <Checkbox
```
Comment (pavyarov) [inline src/modules/manage-sessions-pane/manage-new-session/manage-new-session.tsx:81]: `Fields.Checkbox`

File: src/modules/manage-sessions-pane/manage-sessions-pane.tsx [hunk 5]
```diff
@@ -47,26 +47,38 @@ export const ManageSessionsPane = manageSessionsPane(
...
                 return response;
               } catch (error) {
                 if (error?.response?.data?.code === 409) {
                   const { data: { fieldErrors = [] } = {} } = error?.response?.data || {};
-                  return handleFieldErrors(fieldErrors);
+                  const fieldErrorsToObject = handleFieldErrors(fieldErrors);
+                  if (values.isGlobal) {
+                    showGeneralAlertMessage({
+                      type: 'ERROR',
+                      text: fieldErrorsToObject?.sessionId,
```
Comment (pavyarov) [inline src/modules/manage-sessions-pane/manage-sessions-pane.tsx:76]: @RomanAchilov Some strange logic is here. Plz check it before merge.

**Review by pavyarov — CHANGES_REQUESTED** [review]

File: src/modules/manage-sessions-pane/manage-sessions-pane.tsx (hunk 5 above)
Comment (RomanAchilov) [reply src/modules/manage-sessions-pane/manage-sessions-pane.tsx:76]: yes

File: src/modules/manage-sessions-pane/manage-new-session/manage-new-session.tsx (hunk 3 above)
Comment (SMPlay) [reply src/modules/manage-sessions-pane/manage-new-session/manage-new-session.tsx:57]: If I'll use simple Fields.Checkbox then I can't use styles for checkboxes

File: src/modules/manage-sessions-pane/manage-new-session/manage-new-session.module.scss [hunk 6]
```diff
@@ -16,4 +16,25 @@
...
+
+  &__checkboxes {
+    margin-top: 16px;
+  }
+
+  &__checkbox {
+    margin-right: 8px;
+
+    &:not(:first-child) {
+      margin-top: 16px;
```
Comment (IliaSolovev) [inline src/modules/manage-sessions-pane/manage-new-session/manage-new-session.module.scss:28]: can you use flex-box in parent block instead it?

File: src/modules/manage-sessions-pane/manage-sessions-pane-api/start-agent-session.ts [hunk 7]
```diff
@@ -1,13 +1,15 @@
 import axios from 'axios';
 
+import { StartSessionPayload } from 'types/start-session-payload';
+
 export function startAgentSession(
   agentId: string,
   pluginId: string,
 ) {
-  return async ({ sessionId }: {sessionId: string}): Promise<void> => {
+  return async ({ sessionId, isGlobal = false, isRealtime = false }: StartSessionPayload): Promise<void> => {
```
Comment (IliaSolovev) [inline src/modules/manage-sessions-pane/manage-sessions-pane-api/start-agent-session.ts:9]: default values to these params is necessary?  Maybe BE have default value?

**Review by IliaSolovev — CHANGES_REQUESTED** [review]

File: src/modules/manage-sessions-pane/manage-new-session/manage-new-session.module.scss [hunk 8]
```diff
@@ -16,4 +16,25 @@
   &__new-session-form {
     padding: 16px 24px;
   }
+
+  &__checkboxes {
+    display: flex;
```
Comment (RomanAchilov) [inline src/modules/manage-sessions-pane/manage-new-session/manage-new-session.module.scss:21]: panel

File: src/modules/manage-sessions-pane/manage-new-session/manage-new-session.module.scss [hunk 9]
```diff
@@ -16,4 +16,25 @@
   &__new-session-form {
```
Comment (RomanAchilov) [inline src/modules/manage-sessions-pane/manage-new-session/manage-new-session.module.scss:16]: `
display: grid
gap: 16px
`

Comment (RomanAchilov) [reply src/modules/manage-sessions-pane/manage-new-session/manage-new-session.module.scss:16]: or display: flex
flex-dir: col
gap: 16

Comment (RomanAchilov) [reply src/modules/manage-sessions-pane/manage-new-session/manage-new-session.module.scss:16]: it is not necessary to use a margin inside the content blocks for indents

File: src/modules/manage-sessions-pane/manage-new-session/manage-new-session.module.scss [hunk 10]
```diff
@@ -16,4 +16,25 @@
...
     padding: 16px 24px;
   }
+
+  &__checkboxes {
+    display: flex;
+    flex-direction: column;
+    justify-content: space-between;
+    height: 56px;
+    margin-top: 16px;
```
Comment (RomanAchilov) [inline src/modules/manage-sessions-pane/manage-new-session/manage-new-session.module.scss:25]: it is not necessary to use a margin inside the content blocks for indents

File: src/modules/manage-sessions-pane/manage-new-session/manage-new-session.tsx [hunk 11]
```diff
@@ -42,9 +48,51 @@ export const ManageNewSession = manageNewSession(({ className, agentId, serviceG
...
+                label="Set as global session"
+                input={input}
+                meta={meta}
+              />
+              <Tooltip
+                message={(
+                  <TooltipMessage>
+                    <div>
+                      Session that tracks all of the executions on your JVM
+                    </div>
```
Comment (RomanAchilov) [inline src/modules/manage-sessions-pane/manage-new-session/manage-new-session.tsx:67]: 1 line

File: src/modules/manage-sessions-pane/manage-new-session/manage-new-session.tsx (hunk 3 above)
Comment (RomanAchilov) [reply src/modules/manage-sessions-pane/manage-new-session/manage-new-session.tsx:57]: u can

File: src/modules/manage-sessions-pane/manage-sessions-pane.tsx [hunk 12]
```diff
@@ -47,25 +47,38 @@ export const ManageSessionsPane = manageSessionsPane(
...
+                form.change('isGlobal', false);
+                form.change('isRealtime', false);
                 setIsNewSession(false);
                 return response;
               } catch (error) {
                 if (error?.response?.data?.code === 409) {
-                  const { data: { fieldErrors = [] } = {} } = error?.response?.data || {};
+                  const { data: { fieldErrors = [] } = {}, message = '' } = error?.response?.data || {};
+
+                  if (fieldErrors.length === 0) {
```
Comment (RomanAchilov) [inline src/modules/manage-sessions-pane/manage-sessions-pane.tsx:73]: ?

**Review by RomanAchilov — CHANGES_REQUESTED** [review]

File: src/modules/manage-sessions-pane/manage-new-session/manage-new-session.tsx [hunk 13]
```diff
@@ -40,11 +46,52 @@ export const ManageNewSession = manageNewSession(({ className, agentId, serviceG
           )}
       </span>
     </GeneralAlerts>
-    <NewSessionForm label="Session ID">
-      <Field name="sessionId" component={Fields.Input} placeholder="Enter session ID" />
+    <NewSessionForm>
+      <FormGroup label="Session ID">
+        <Field name="sessionId" component={Fields.Input} placeholder="Enter session ID" />
+      </FormGroup>
+      <Checkboxes direction="column" align="space-between" verticalAlign="start">
```
Comment (RomanAchilov) [inline src/modules/manage-sessions-pane/manage-new-session/manage-new-session.tsx:53]: useless

File: src/modules/manage-sessions-pane/manage-new-session/manage-new-session.module.scss [hunk 14]
```diff
@@ -14,6 +14,21 @@
   }
 
   &__new-session-form {
+    display: grid;
+    gap: 16px;
     padding: 16px 24px;
   }
+
+  &__checkboxes {
```
Comment (RomanAchilov) [inline src/modules/manage-sessions-pane/manage-new-session/manage-new-session.module.scss:22]: useless

Comment (RomanAchilov) [reply src/modules/manage-sessions-pane/manage-new-session/manage-new-session.module.scss:22]: please think about how to write styles optimally

File: src/modules/manage-sessions-pane/manage-new-session/manage-new-session.tsx (hunk 13 above)
Comment (RomanAchilov) [reply src/modules/manage-sessions-pane/manage-new-session/manage-new-session.tsx:53]: please think about how to write styles optimally

File: src/modules/manage-sessions-pane/manage-new-session/manage-new-session.tsx [hunk 15]
```diff
@@ -40,11 +46,52 @@ export const ManageNewSession = manageNewSession(({ className, agentId, serviceG
...
+        />
+      </Checkboxes>
     </NewSessionForm>
   </div>
 ));
 
 const SettingsLink = manageNewSession.settingsLink(NavLink);
-const NewSessionForm = manageNewSession.newSessionForm(FormGroup);
+const NewSessionForm = manageNewSession.newSessionForm('div');
+const Checkboxes = manageNewSession.checkboxes(Panel);
```
Comment (RomanAchilov) [inline src/modules/manage-sessions-pane/manage-new-session/manage-new-session.tsx:95]: delete. U use grid layout

File: src/modules/manage-sessions-pane/manage-sessions-pane-api/start-agent-session.ts [hunk 16]
```diff
@@ -1,13 +1,15 @@
 import axios from 'axios';
 
+import { StartSessionPayload } from './start-session-payload';
```
Comment (RomanAchilov) [inline src/modules/manage-sessions-pane/manage-sessions-pane-api/start-agent-session.ts:3]:
```suggestion
import { StartSessionPayload } from './start-session-payload-types';
```

File: src/modules/manage-sessions-pane/manage-new-session/manage-new-session.module.scss [hunk 17]
```diff
@@ -14,6 +14,21 @@
...
+    height: 56px;
+  }
+
+  &__global-session-checkbox {
+    width: 184px;
```
Comment (RomanAchilov) [inline src/modules/manage-sessions-pane/manage-new-session/manage-new-session.module.scss:27]: gap: 8px

File: src/modules/manage-sessions-pane/manage-new-session/manage-new-session.tsx [hunk 18]
```diff
@@ -40,11 +46,52 @@ export const ManageNewSession = manageNewSession(({ className, agentId, serviceG
...
+    <NewSessionForm>
+      <FormGroup label="Session ID">
+        <Field name="sessionId" component={Fields.Input} placeholder="Enter session ID" />
+      </FormGroup>
+      <Checkboxes direction="column" align="space-between" verticalAlign="start">
+        <Field
+          name="isGlobal"
+          type="checkbox"
+          render={({ input, meta }) => (
+            <GlobalSessionCheckbox align="space-between">
```
Comment (RomanAchilov) [inline src/modules/manage-sessions-pane/manage-new-session/manage-new-session.tsx:58]:
```suggestion
            <GlobalSessionCheckbox>
```

**Review by RomanAchilov — CHANGES_REQUESTED** [review]

**Pull request merged**